File: pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>
#include <ctype.h>
#include <string.h>

int main(void)
{
    string text = get_string("Text: ");

    int letters = 0;
    int words = 0;
    int sentences = 0;

    for (int a = 0, len = strlen(text); a < len; a++)
    {
        if (isalnum(text[a]))
        {
            letters++;
        }
        else if (text[a] == ' ')
        {
            words++;
        }
        else if (text[a] == '.' || text[a] == '?' || text[a] == '!')
        {
            sentences++;
        }
    }
    
    float L = (float) letters * 100 / ((float) words + 1);
    float S = (float) sentences * 100 / ((float) words + 1);
    float index = 0.0588 * L - 0.296 * S - 15.8;
    index = round(index);
    
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {   
        printf("Grade %i\n", (int)index);
    }
}

File: pset6/readability/readability.py
# import get_string from cs50 library
from cs50 import get_string

# prompt the user for text
text = get_string("Text: ")

# initialize the three variables
letters = 0
words = 0
sentences = 0

# iterate trough all of the character of the string
for a in range(0, len(text), 1):
    if text[a].isalnum() is True:
        letters += 1
    elif text[a] == " ":
        words += 1
    elif text[a] == "." or text[a] == "?" or text[a] == "!":
        sentences += 1
        
# some formula for the index/grade
L = float(letters) * 100 / (float(words) + 1)
S = float(sentences) * 100 / (float(words) + 1)
index = 0.0588 * L - 0.296 * S - 15.8
index = round(index)

# conditions 
if index < 1:
    print("Before Grade 1")
elif index > 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")